### PR TITLE
adding semant for built-in "print"

### PR DESCRIPTION
**Pull request by rxie25**
fixes issue #10

### CODE REVIEW

**Comment by jzacsh** [comment]
for `print` not `printf`, right?

**Comment by rxie25** [comment]
All our tests had printf, so are we moving towards print now?

**Comment by jzacsh** [comment]
:D Not *all* our tests

```sh
grep -rn print\( tests/*.eq | wc -l
51
$ grep -rn printf\( tests/*.eq | wc -l
12
```
Just 12 lines in [**one** file](https://github.com/rxie25/PLT2016Spring/commit/a94643b7ab290678f0c9) that issue #10 hopefully fixes. :P

**Comment by rxie25** [comment]
ok, I can change it to `print`

**Comment by rxie25** [comment]
@jzacsh @nnhoang take a quick look. Let me know what you guys think.

File: src/semant.ml [hunk 1]
```diff
@@ -118,7 +122,11 @@ let check (contexts, finds) =
               | A.Binop(left, op, right) -> ()
               | A.Unop(op, expr) -> ()
               | A.Assign(left, expr) -> ()
-              | A.Builtin(name, expr) -> ()
+              | A.Builtin(name, expr) -> (
+                match name with
+                    | "print" -> ()
```
Comment (jzacsh) [inline src/semant.ml]: can you check the [TODOs in the codebase](https://github.com/rxie25/PLT2016Spring/tree/master/src#status-), there's a TODO about more builtins we should be supporting, and you can add them right here in this pull request ot this match statement.

File: src/semant.ml [hunk 2]
```diff
@@ -64,7 +64,16 @@ let check (contexts, finds) =
...
+                    | "cos" -> ()
+                    | "sin" -> ()
+                    | "sqrt" -> ()
+                    | "tan" -> ()
+                    | "log" -> ()
+                    | _ -> fail ("incorrect build-in function, " ^ quot name)
```
Comment (jzacsh) [inline src/semant.ml]: nit: use message "unknown" (not "incorrect")

File: src/semant.ml (hunk 1 above)
Comment (jzacsh) [inline src/semant.ml]: you have this exact block of code called in two places. can you factor it out into a new function? eg: `check_builtin`

Comment (rxie25) [inline src/semant.ml]: added builtin keyboards from LRM. I'll open a separate semantic-check pull request for them.

**Comment by rxie25** [comment]
take a look and merge this one.
I'll will continue in pull request #74

File: src/semant.ml (hunk 1 above)
Comment (rxie25) [inline src/semant.ml]: sure, I'll put them together.

Comment (jzacsh) [inline src/semant.ml]: I don't see the new `check_builtin` method anywhere. Did you forget to push? Or am i looking too early?

Comment (rxie25) [inline src/semant.ml]: just updated it.

**Pull request closed without merging**